pkg/kube/krt/fetch.go

```diff
@@ -0,0 +1,58 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package krt
+
+func FetchOne[T any](ctx HandlerContext, c Collection[T], opts ...FetchOption) *T {
```

Comment:
I think the return type needs to be of type T, rather than a pointer to a T.  Otherwise you get pointers to pointers in this less than ideal scenario:

```
// kclient.New cannot allow T = gateway.GatewayClass (without pointer) due to interface receivers
gatewayClasses := kclient.New[*gateway.GatewayClass](client)
kgwc := krt.WrapClient(gatewayClasses)
gwc := krt.FetchOne(ctx, kgwc, krt.FilterName("foo", ""))
// gwc is of type **gateway.GatewayClass
```

Reply:
IMO it cannot just be `T` since that doesn't give indication if it exists. It could be `(T, bool)` but that is a huge pain to use